Localization/TranslationService.cs

```diff
@@ -574,11 +574,25 @@ public string GetMatchingIfExists(string key, params object[] args)
                 .FirstOrDefault();
         }
 
-        private void LoadEntries(Dictionary<string, string> entryCache, Dictionary<string, KeyValuePair<Regex, string>> regexCache, Dictionary<string, string> wildcardCache, Dictionary<string, string> entries, string domain = GlobalConstants.DefaultDomain)
+        private void LoadEntries(Dictionary<string, string> entryCache, Dictionary<string, KeyValuePair<Regex, string>> regexCache, Dictionary<string, string> wildcardCache, JToken json, string domain = GlobalConstants.DefaultDomain, string key = "")
         {
-            foreach (var entry in entries)
+            switch (json)
             {
-                LoadEntry(entryCache, regexCache, wildcardCache, entry, domain);
+                case JObject jsonObject:
+                    foreach (var property in jsonObject.Properties())
+                    {
+                        var newKey = key.Length == 0
+                            ? property.Name
+                            : $"{key}-{property.Name}";

```

Comment:
Since it is to be expected that there will be a lot of translation keys that need to be concated, and the key prefix can be reused for all descendants, please use a `StringBuilder` here instead of the `key` function argument.

The idea here would be to do something like 

```
if (key.Length > 0)   key.Append('-')
var prefixLength = key.Length;
foreach(var property in ...) {
  key.Length = prefixLength; // reset to prefix
  key.Append(property.Name); // append suffix
  ... // descend deeper
}
```